lm_eval/logging_utils.py

```diff
@@ -0,0 +1,380 @@
+
+        Args:
+            data (List[Dict[str, Any]]): The data to generate a dataset for.
+            config (Dict[str, Any]): The configuration of the task.
+
+        Returns:
+            pd.DataFrame: A dataframe that is ready to be uploaded to W&B.
+        """
+        ids = [x["doc_id"] for x in data]
+        labels = [x["target"] for x in data]
+        instance = [""] * len(ids)
+
+        metrics_list = config["metric_list"]
+        metrics = {}
+        for metric in metrics_list:
+            metric = metric.get("metric")
+            metrics[metric] = [x[metric] for x in data]
+
+        if config["output_type"] == "loglikelihood":
+            instance = [x["arguments"][0][0] for x in data]
+            labels = [x["arguments"][0][1] for x in data]
+        elif config["output_type"] == "multiple_choice":
+            instance = [
+                x["arguments"][0][0]
+                + "\n\n"
+                + "\n".join([f"- {y[1]}" for y in x["arguments"]])
+                for x in data
+            ]
+        elif config["output_type"] == "loglikelihood_rolling":
+            instance = [x["arguments"][0][0] for x in data]
+        elif config["output_type"] == "generate_until":
+            instance = [x["arguments"][0][0] for x in data]
+
+        df_data = {
+            "id": ids,
+            "data": instance,
+            "input_len": [len(x) for x in instance],
+            "labels": labels,
+            "output_type": config["output_type"],
+        }
+        df_data.update(metrics)
+
+        return pd.DataFrame(df_data)
+
+    def log_eval_samples(self, samples: Dict[str, List[Dict[str, Any]]]) -> None:
+        """Log evaluation samples to W&B.
+
+        Args:
+            samples (Dict[str, List[Dict[str, Any]]]): Evaluation samples for each task.
+        """
+        for task_name in self.task_names:
+            eval_preds = samples[task_name]
+
+            # log the samples as an artifact
+            dumped = json.dumps(
+                eval_preds,
+                indent=2,
+                default=_handle_non_serializable,
+                ensure_ascii=False,
+            )
+            artifact = wandb.Artifact(f"{task_name}", type="samples_by_task")
+            with artifact.new_file(
+                f"{task_name}_eval_samples.json", mode="w", encoding="utf-8"
+            ) as f:
+                f.write(dumped)
+            self.run.log_artifact(artifact)
+            artifact.wait()
+
+            # log the samples as a W&B Table
+            df = self.generate_dataset(eval_preds, self.task_configs.get(task_name))
+            self.run.log({f"{task_name}_eval_results": df})
+
+    def log_results_as_json(self) -> None:
+        """Log results as JSON artifact to W&B."""
+        dumped = json.dumps(
+            self.results, indent=2, default=_handle_non_serializable, ensure_ascii=False
+        )
+        artifact = wandb.Artifact("results", type="eval_results")
```

Comment:
here we are logging the results.json file to W&B Artifacts. it would be better to use some unique identifier something like - `model-wand_run_id-results`. 

What would be the best way to get the model name from `model` and `model_args` arguments?

Reply:
Could these be passed as args into the `log_eval_result` and subsequently this function?

https://github.com/EleutherAI/lm-evaluation-harness/blob/bfbd0325d3beea7482b866919bc6d1ab05dd7578/lm_eval/evaluator.py#L211-L212

you could combine `model` and `model_args`, perhaps, though that might not be the cleanest: for huggingface models `results["model"]` from the output of simple_evaluate will be the `pretrained` value / modelname, while for other classes this might be passed in via `engine` or `model` kwargs to the LM subclass.

Not sure if this answers your question!